distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso

```diff
@@ -788,6 +872,78 @@ type Integer
+
+       For integers, rounding to 0 or more decimal places simply returns the
+       argument. For negative decimal places, see below.
+
+       By default, rounding uses "asymmetric round-half-up", also known as
+       "round towards positive infinity." If use_bankers=True, then it uses
+       "round-half-even", also known as "banker's rounding".
+
+       Arguments:
+       - decimal_places: The number of decimal places to round to. Can be
+         negative, which results in rounding to positive integer powers of 10.
+         Must be between -15 and 15 (inclusive).
+       - use_bankers: Use banker's rounding
+
+       ! Error Conditions
+         Throws `Illegal_Argument` if the number is 15 or more decimal places.
+         Above 14 digits, it is possible that the underlying long, converted to
+         double in the rounding process, would lose precision in the least
+         significant bits.
+         (See https://en.wikipedia.org/wiki/Double-precision_floating-point_format.)
+
+         If `decimal_places` is outside the range -15..15 (inclusive), an
+         `Illegal_Argument` error is thrown.
+
+       ? Negative decimal place counts
+         Rounding to `n` digits can be thought of as "rounding to the nearest
+         multiple of 10^(-n)". For negative decimal counts, this results in
+         rounding to the nearest positive integer power of 10.
+
+       ? Banker's rounding
+         Banker's rounding rounds half-way values to the nearest even number.
+
+       > Example
+         Round an integer (returns the value unchanged).
+
+            3 . round
+            # ==> 3
+
+       > Example
+         Round to the nearest thousand.
+
+            2511 . round -3
+            # ==> 3000
+
+       > Example
+         Round to the nearest hundred, using Banker's Rounding.
+
+            12250 . round -2 use_bankers=True
+            # ==> 12200
+    round : Integer -> Integer ! Illegal_Argument
+    round self decimal_places=0 use_bankers=False =
+        case decimal_places < round_min_decimal_places || decimal_places > round_max_decimal_places of
+            True ->
+                msg = "round: decimal_places must be between " + round_min_decimal_places.to_text + " and " + round_max_decimal_places.to_text + "(inclusive), but was " + decimal_places.to_text
+                Error.throw (Illegal_Argument.Error msg)
+            False ->
+                case self < round_min_long || self > round_max_long of
+                    True ->
+                        msg = "Error: Integer.round can only accept values between " + round_min_long.to_text + " and " + round_max_long.to_text + "(inclusive), but was " + self.to_text
+                        Error.throw (Illegal_Argument.Error msg)
+                    False ->
+                        ## It's already an integer so unless decimal_places is
+                           negative, the value is unchanged.
+                        if decimal_places >= 0 then self else
```

Comment:
I'd move this check upwards, as we can cope with out-of-range self if doing a no-op.